pkg/expression/builtin_control.go

```diff
@@ -198,8 +221,8 @@ func InferType4ControlFuncs(ctx sessionctx.Context, funcName string, args ...Exp
 		tempFlag := resultFieldType.GetFlag()
 		types.SetTypeFlag(&tempFlag, mysql.NotNullFlag, false)
 		resultFieldType.SetFlag(tempFlag)
-		// If both arguments are NULL, make resulting type BINARY(0).
-		resultFieldType.SetType(mysql.TypeString)
+
+		resultFieldType.SetType(mysql.TypeNull)
```

Comment:
Is this behavior the same as MySQL?

Reply:
> Is this behavior the same as MySQL?

Yes, when we execute `select coalesce(null,null);` the return type is:
```
Field   1:  `coalesce(null,null)`
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       NULL
Collation:  binary (63)
Length:     0
Max_length: 0
Decimals:   0
Flags:      BINARY NUM
```